### ISSUE TITLE
1st setup: Broken Helper

### ISSUE THREAD

**Issue by unlucio** [tetele/hvac_group#54]
## System Information

version | core-2024.1.5
-- | --
installation_type | Home Assistant Supervised
dev | false
hassio | true
docker | true
user | root
virtualenv | false
python_version | 3.11.6
os_name | Linux
os_version | 5.10.0-21-arm64
arch | aarch64
timezone | Europe/Rome
config_dir | /config

<details><summary>Home Assistant Community Store</summary>

GitHub API | ok
-- | --
GitHub Content | ok
GitHub Web | ok
GitHub API Calls Remaining | 4479
Installed Version | 1.34.0
Stage | running
Available Repositories | 1391
Downloaded Repositories | 56

</details>

<details><summary>Home Assistant Cloud</summary>

logged_in | false
-- | --
can_reach_cert_server | ok
can_reach_cloud_auth | ok
can_reach_cloud | ok

</details>

<details><summary>Home Assistant Supervisor</summary>

host_os | Debian GNU/Linux 11 (bullseye)
-- | --
update_channel | stable
supervisor_version | supervisor-2023.12.1
agent_version | 1.4.1
docker_version | 23.0.4
disk_total | 457.4 GB
disk_used | 201.2 GB
healthy | true
supported | failed to load: Unsupported
supervisor_api | ok
version_api | ok
installed_addons | ESPHome (2023.12.9), Mosquitto broker (6.4.0), Piper (1.4.0), Whisper (1.0.2), openWakeWord (1.8.2), Studio Code Server (5.15.0)

</details>

<details><summary>Dashboards</summary>

dashboards | 5
-- | --
resources | 29
views | 27
mode | storage

</details>

<details><summary>Recorder</summary>

oldest_recorder_run | 3 February 2023 at 18:17
-- | --
current_recorder_run | 29 January 2024 at 17:43
estimated_db_size | 9242.86 MiB
database_engine | mysql
database_version | 10.6.11

</details>

<details><summary>Sonoff</summary>

version | 3.5.4 (a4a8c5f)
-- | --
cloud_online | 0 / 8
local_online | 8 / 8
debug | failed to load: 

</details>

<details><summary>Spotify</summary>

api_endpoint_reachable | ok
-- | --

</details>

### Checklist

- [X] I have enabled debug logging for my installation.
- [X] I have filled out the issue template to the best of my ability.
- [X] This issue only contains 1 issue (if you have multiple issues, open one issue for each issue).
- [X] This issue is not a duplicate issue of currently [previous issues](https://github.com/tetele/hvac_group/issues?q=is%3Aissue+label%3A%22Bug%22+)..

### Describe the issue

- Installed the integration form HACS
- Devices & Services -> Add Integration -> HVAC group
- filled in the required data
- ended up with a broken, undeletable, helper:
<img width="489" alt="Screenshot 2024-01-29 at 17 42 26" src="https://github.com/tetele/hvac_group/assets/778799/7987a750-ceea-4913-a1cc-141d9fa61550">
<img width="1702" alt="Screenshot 2024-01-29 at 17 41 53" src="https://github.com/tetele/hvac_group/assets/778799/f6e65752-7d5c-4b17-a308-1620fbc1b596">

### Reproduction steps

- Installed the integration form HACS
- Devices & Services -> Add Integration -> HVAC group
- filled in the required data
- enjoy broken helper
(I've no idea, I didn't try over and over just to run the chance of accumulating other broken helpers, honestly, this thing is requiring way more time than I bargained for, I just want to get rid of it)

### Debug logs

```text
Logs would be worthless: I checked this out yesterday night before going to bed, and when it ended up in a broken helper I just called it a night.
Right now those logs are gone (new docker container), and I'm not gonna try again, just to produce more logs, and end up with another broken helper I cannot delete. Having one is already annoying enough.
```

### Diagnostics dump

🤷‍♂️

**Comment by tetele** [comment]
I apologize for any inconvenience. I can assure you nobody wants to put bad integrations out there to mess with people's patience. Or at least i don't.

The reason this issue template requests logs is that, without them, i can't understand the cause for which it is behaving the way it is and therefore i can't fix it. If i were able to reproduce the behavior you described, i would have fixed it before publishing a version.

I've been working on fixing some other issues that may be connected to this, so please wait a bit for a possible fix. In the meantime, if you're willing to help, error logs would be great.

Thanks!

**Comment by unlucio** [comment]
I'm sorry if the ticket seems a bit direct:
I do well know and understand the value of logs, but unfortunately, they not always be available, and with a mandatory field I had to write something in it (in this case I did update HA in the hope that that would have fixed the broken sensor, resulting in not fixing it and deleting the logs).
SW is made by humans, and bugs are bound to happen, that's fine.
My frustration didn't come from the bug but from the inability to delete the broken group.
It's just a matter of user experience: tinkering can be fun when one is up for tinkering, but some other times the mood might just be "let's check this out", great if it works, fine if it doesn't, and in that case imho it is important to leave the option to quickly clean everything up.
Clicking on the broken group just presented the settings inputted, adding a delete button would have prevented frustration and given the chance to produce more logs, if one is up for it, without the risk of ending up with several broken helpers, and still having the option to bail out if they're not up to it.
All good, at the end I manually cleaned up my HA and got rid of the broken helper.

**Comment by tetele** [comment]
I've almost completely re-written the integration and I'm hoping it's much more robust now. There's a [0.2.0-beta](https://github.com/tetele/hvac_group/releases/tag/0.2.0-beta.1) version you could try (but you have to enable beta versions in HACS).

This issue was most likely fixed in #62 but please reopen it if it's not. I know you just wanted to get rid of the integration, though, so I totally get it if you won't try :)
